homeassistant/components/recorder/models.py

```diff
@@ -223,7 +223,20 @@ def to_native(self, validate_entity_id=True):
             return None
 
 
-class StatisticData(TypedDict, total=False):
+class StatisticResult(TypedDict):
+    """Statistic data class."""
+
+    meta: StatisticMetaData
+    stat: tuple[StatisticData, ...]
```

Comment:
Why is this a tuple?

Reply:
It should be an iterable.
The purpose is to allow inserting multiple datapoints for unchanged metadata, for example a day's or month's worth of hourly updates and not have to lookup the metadata already stored in the database for each point.